temporal-sdk/src/main/java/io/temporal/common/context/ContextPropagator.java

```diff
@@ -126,4 +130,27 @@
 
   /** Sets the current context */
   void setCurrentContext(Object context);
+
```

Comment:
Looking at this and the implementation I'm getting more inclined to just using interceptors for this integration and getting rid of the context propagator interface completely as it has very limited use.

Reply:
I’d be happy to look into that - are there any docs or examples of interceptors anywhere? I didn’t see a way of propagating information from worker/client to the server and back using interceptors.

Reply:
I think you are right that there is something missing in the interceptors to support this use case. Let me think about this.